src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -76,7 +76,7 @@ def trace(self, append_to_templated: str = "") -> JinjaTrace:
         trace_template = self.make_template(trace_template_str)
         trace_template_output = trace_template.render()
         # Split output by section. Each section has two possible formats.
-        trace_entries = list(regex.finditer(r"\0", trace_template_output))  # type: ignore[call-overload] # noqa: E501
+        trace_entries = list(regex.finditer(r"\0", trace_template_output))
```

Comment:
This recent mypy issue has now been fixed by the release of `types-regex` version 2022.10.31.1.